src/main/java/com/jvavateam/carsharingapp/repository/SpecificationProvider.java

```diff
@@ -0,0 +1,12 @@
+package com.jvavateam.carsharingapp.repository;
+
+import com.jvavateam.carsharingapp.model.Rental;
+import org.springframework.data.jpa.domain.Specification;
+
+public interface SpecificationProvider<T> {
+    String getKey();
+
+    Specification<Rental> getSpecification(Boolean params);
+
+    Specification<Rental> getSpecification(Long params);
+}
```

Comment:
Can we make `getSpecification()` method generic? What if we want to search by rental date?